src/main/scala/org/allenai/plugins/CoreSettingsPlugin.scala

```diff
@@ -141,6 +150,71 @@ object CoreSettingsPlugin extends AutoPlugin {
   // See http://www.scala-sbt.org/release/docs/Testing.html#Custom+test+configuration
   override val projectConfigurations = Seq(Configurations.IntegrationTest extend (Test))
 
+  /** Scalariform settings we use that are different from the defaults */
+  val ScalariformDefaultOverrides: Seq[(PreferenceDescriptor[_], Boolean)] = Seq(
+    (DoubleIndentClassDeclaration, true),
+    (MultilineScaladocCommentsStartOnFirstLine, true),
+    (PlaceScaladocAsterisksBeneathSecondAsterisk, true),
+    (SpacesAroundMultiImports, true)
+  )
+
+  // These settings will be automatically applied to the build exactly once and will not be applied
+  // to individual subprojects. Tasks added to buildSettings will _not_ show up in tab-completion in
+  // the SBT console. However, this is a good place to put tasks that you do not want executed once
+  // per subproject.
+  override def buildSettings: Seq[Setting[_]] = Seq(
+    generateAutoformatGitHook := {
+      val expectedGitHooksDir = (baseDirectory in ThisBuild).value / ".git" / "hooks"
+      val preCommitFile = expectedGitHooksDir / "pre-commit"
+      val scalariformFile = expectedGitHooksDir / "scalariform.jar"
+      def requireFilesDontExist(files: File*) = {
+        files find (_.exists) foreach { file =>
+          sys.error(s"You already have .git/hooks/${file.getName}. Remove or rename the file and run again.")
+        }
+      }
+      requireFilesDontExist(preCommitFile, scalariformFile)
+
+      def withResource[A](resource: String)(f: InputStream => A) = {
```

Comment:
What about using sbt's `IO.readLinesURL` and `IO.writeLines` instead of writing these helpers?